Per-task aspect schema & layout querying

mg6 opened the pull request:
Extends existing set of worker proxy calls:

- `GET /schema/outputs/<string:task_name>/`
- `GET /layout/outputs/<string:task_name>/`

with per-aspect calls:

- `GET /schema/outputs/<string:task_name>/<string:aspect_name>`
- `GET /layout/outputs/<string:task_name>/<string:aspect_name>`

gmrukwa commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create master node for distribution of calculations](https://trello.com/c/g4lghTpW/331-create-master-node-for-distribution-of-calculations)

gmrukwa commented on README.md:
```diff
@@ -142,6 +145,9 @@ Sample JSON for such styling:
 Format of `layout` should be as samples included [here](https://angular2-json-schema-form.firebaseapp.com/).
 
+You may also query for single aspect instead of complete list,
+similarly as for `/schema` endpoint: `/layout/inputs/<task-name>/<aspect-name>`.
```
`/layout/outputs/...`

BTW. right now it asks for `/layout/outputs/aspect-name/task-name/`.

mg6 replied:
Fixed.

gmrukwa commented on api.py:
```diff
@@ -61,6 +61,16 @@ def get_outputs(task_name: str) -> Response:
+    Returns:
+        Normalized output query pattern definition.
+    """
+    return _ask_for(task_name, "/schema/outputs/{aspect_name}".format(aspect_name=aspect_name))
```
This way format will be reversed: to know, how to get to `/schema/outputs/aspect_name/task_name/` we will have to ask for `/schema/outputs/task_name/`. Can we make this straight?

mg6 replied:
Refactored the entire proxying logic to fix the problem you described. Please check commit 9a47973.

gmrukwa replied:
We are still missing task name in the endpoint url.

mg6 committed:
Refactor proxying logic to expect full endpoints

As of now, target endpoint has to be fully specified when proxying.

This removes the limitation of composing proxied requests only with
specific parameter (task name) at the end of the endpoint. Such a change
was needed to account for multiple parameters from Flask route handler
(task name & aspect name).

Additionally, _proxy and _ask_for methods were renamed _jsonify and
_proxy, respectively, so that they better describe what happens inside.

_proxy method was extended to support simple POST request proxying, in
addition to already used GET requests. One can provide the desired
request handler through `method` parameter. Additional parameters for
the handler can be passed through *args and **kwargs.

gmrukwa commented on api.py:
```diff
@@ -78,7 +88,17 @@ def get_outputs_layout(task_name: str) -> Response:
+    Returns:
+        Definition of form for narrowing down the result scope.
+    """
+    return _ask_for(task_name, "/layout/outputs/{aspect_name}".format(aspect_name=aspect_name))
```
We are missing task name in this url.

KapRom reviewed (dismissed):
No remarks from me whatsoever.

mg6 committed:
Support per-aspect schema & layout querying

mg6 committed:
Document per-aspect schema & layout endpoints

Fix incorrect output aspect querying doc
